dap-mode.el

```diff
@@ -1598,6 +1621,49 @@ point is set."
   (interactive)
   (dap-mode t))
 
+
+;; Auto configure
+
+(defun dap--show-feature-window (_session)
+  "Show auto confiured feature windows."
+  (seq-doseq (feature-start-stop dap-auto-configure-features)
+    (when-let ((start-stop (alist-get feature-start-stop dap-feature->start-stop)))
+      (funcall (car start-stop)))))
+
+(defun dap--hide-feature-window (_session)
+  "Hide all debug windows when sessions are dead."
+  (unless (-filter 'dap--session-running (dap--get-sessions))
+    (seq-doseq (feature-start-stop dap-auto-configure-features)
+      (-when-let* ((feature-start-stop (alist-get feature-start-stop dap-feature->start-stop))
+                   (buffer-name (symbol-value (cdr feature-start-stop))))
+        (and (get-buffer buffer-name)
+             (kill-buffer buffer-name))))))
+
+(define-minor-mode dap-auto-configure-mode
+  "Auto configure dap minor mode."
+  :init-value nil
+  :group 'dap-mode
+  (cond
+   (dap-auto-configure-mode
+    (dap-mode 1)
+    (dap-ui-mode 1)
+    (seq-doseq (feature dap-auto-configure-features)
+      (when-let ((mode (alist-get feature dap-feature->mode)))
+        (if (consp mode)
+            (when (require (cdr mode) nil t)
+              (funcall (car mode) 1))
+            (funcall mode 1))))
+    (add-hook 'dap-stopped-hook #'dap--show-feature-window)
+    (add-hook 'dap-terminated-hook #'dap--hide-feature-window))
+   (t
+    (dap-mode -1)
+    (dap-ui-mode -1)
+    (seq-doseq (feature dap-auto-configure-features)
+      (when-let ((mode (alist-get feature dap-feature->mode)))
+        (funcall mode -1)))
```

Comment:
I think you need to check the `mode` for `consp` here

Reply:
nice catch!